serve/triton/type_gen.py

```diff
@@ -70,30 +71,41 @@ def tensor2ndarray(placehoder: str):
     return placehoder + '.as_numpy()'
 
 
+AttrInfo = namedtuple('AttrInfo', ['tensor_placeholder', 'obj_placeholder', 'shape', 'triton_dtype', 'numpy_dtype'])
+
+
+class TypeInfo:
+    def __init__(self, attr_info, t, shape, is_list=False):
+        self.attr_info = attr_info
+        self.type = t
+        self.shape = shape
+        self.is_list = is_list
+
+
 class ImageType:
     """
     A collection of type handling function of `towhee.types.Image`
     """
 
     @staticmethod
-    def unpack_attrs(shape, is_list):
-        if is_list:
-            data_part = ('$obj', (-1, ) + shape, 'TYPE_INT8', is_list)
-            mode_part = ('$obj.mode', (-1, ), 'TYPE_STRING', is_list)
-        else:
-            data_part = ('$obj', shape, 'TYPE_INT8', is_list)
-            mode_part = ('$obj.mode', (), 'TYPE_STRING', is_list)
+    def type_info(t, shape, is_list):
+        data_part = AttrInfo('$t_data', '$obj', shape, 'TYPE_INT8', 'numpy.int8')
+        mode_part = AttrInfo('$t_mode', '$obj.mode', (), 'TYPE_STRING', 'numpy.object_')
```

Comment:
https://github.com/triton-inference-server/python_backend/blob/3a60cfc7fb3525d1200ee179a1355fd813cccd28/src/resources/triton_python_backend_utils.py#L237

Might be useful. How come we are enforing int8?